File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/adapter/ForwardPlayer.java
package jv.com.tencent.cain.design_pattern.adapter;

/**
 * @author cainjiang
 * @date 2020/4/15
 */
public class ForwardPlayer extends Player {

    public ForwardPlayer(String name) {
        super(name);
    }

    @Override
    public void attack() {
        System.out.println("ForwardPlayer# " + name + " attack ---");
    }

    @Override
    public void defense() {
        System.out.println("ForwardPlayer# " + name + " defense ---");
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/adapter/ForwardPlayer.kt
package kt.com.tencent.cain.desigin_pattern.adapter

/**
 * @author cainjiang
 * @date 2020/4/15
 */
class ForwardPlayer(name: String) : Player(name) {

    override fun attack() {
        println("ForwardPlayer# $name attack ---")
    }

    override fun defense() {
        println("ForwardPlayer# $name defense ---")
    }

}